### DISCUSSION TITLE
Enum.HasAnyFlag

### DISCUSSION THREAD

**Discussion by ulfemsoy** [dotnet/runtime#107824] [Ideas]
The current Enum.HasFlag(value, flag) is defined as:
(value & flag) == flag

I suggest to introduce another common method called Enum.HasAnyFlag(value, flag) defined as
(value & flag) != 0

**Comment by huoyaoyuan** [comment]
`Enum.HasFlag` is well-known to be unfriendly to performance. Although there's optimization now, it's still causing excessive allocations under Debug config.
I'm afraid there won't be interest until a better pattern is employed.

**Reply by colejohnson66** [reply]
Isn’t `Enum.HasFlag` intrinsic on 8.0 and up? Theoretically, `Enum.HasAnyFlag` would be as well.

**Reply by huoyaoyuan** [reply]
The intrinsic isn't enabled on Debug application. Someone told me that it's causing excessive allocation.
Moreover, `HasFlag<T>(this T left, T rigit)` can have even better type safety.

**Reply by hez2010** [reply]
It's old news. Now with https://github.com/dotnet/runtime/pull/89348 those allocations are gone even under Debug.

**Reply by huoyaoyuan** [reply]
> Now with #89348 those allocations are gone even under Debug.

Is it? ~Tier 0 is not Debug. The optimization is still disabled under MinOpts.~

Since the PR explicitly mentions Debug, I think it's covered. But I remember seeing `OptimizationDisabled` condition when I check it recently.
